libbeat/outputs/elasticsearch/client.go

```diff
@@ -302,36 +314,35 @@ func bulkEncodePublishRequest(
 	body bulkWriter,
 	index outil.Selector,
 	pipeline *outil.Selector,
-	data []outputs.Data,
-) []outputs.Data {
+	data []publisher.Event,
+) []publisher.Event {
 	okEvents := data[:0]
-	for _, datum := range data {
-		meta := createEventBulkMeta(index, pipeline, datum)
-		if err := body.Add(meta, datum.Event); err != nil {
+	for i := range data {
+		event := &data[i].Content
+		meta := createEventBulkMeta(index, pipeline, event)
+		if err := body.Add(meta, event); err != nil {
 			logp.Err("Failed to encode event: %s", err)
 			continue
 		}
-		okEvents = append(okEvents, datum)
+		okEvents = append(okEvents, data[i])
 	}
 	return okEvents
 }
 
 func createEventBulkMeta(
 	index outil.Selector,
 	pipelineSel *outil.Selector,
-	data outputs.Data,
+	event *beat.Event,
 ) interface{} {
-	event := data.Event
-
-	pipeline, err := getPipeline(data, pipelineSel)
+	pipeline, err := getPipeline(event, pipelineSel)
 	if err != nil {
 		logp.Err("Failed to select pipeline: %v", err)
 	}
 
 	if pipeline == "" {
 		type bulkMetaIndex struct {
-			Index   string `json:"_index"`
-			DocType string `json:"_type"`
+			Index   string `json:"_index" struct:"_index"`
```

Comment:
yes, go-structform being encoding-agnostic uses the `struct` struct-tag by default. For now I preferred the types to be compatible with both, stdlib json and go-structform.